src/main/java/Duke.java

```diff
@@ -1,10 +1,265 @@
+                        tasks[taskNumber - 1].markTaskAsDone();
+                    } else {
+                        tasks[taskNumber - 1].markTaskAsUndone();
+                    }
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    System.out.println("____________________________________________________________\n");
+                    System.out.println("☹ OOPS!!! You haven't entered a task number. Please enter a task number!");
+                    System.out.println("____________________________________________________________");
+                    continue;
+                } catch (InputMismatchException e) {
+                    System.out.println("____________________________________________________________\n");
+                    System.out.println("☹ OOPS!!! You haven't entered a valid task number to mark or unmark. Please enter a task number!");
+                    System.out.println("____________________________________________________________");
+                    continue;
+                } catch (DukeException e) {
+                    System.out.println("____________________________________________________________\n");
+                    System.out.println("☹ OOPS!!! The task number that you have entered is out of range. Please enter a different task number!");
+                    System.out.println("____________________________________________________________");
+                    continue;
+                }
+                System.out.println(tasks[taskNumber - 1]);
+                System.out.println("____________________________________________________________");
+            } else {
+                taskType = splitUserInput[0];
+                try {
+                    checkInputValidity(taskType);
+                } catch(DukeException e) {
+                    System.out.println("____________________________________________________________\n");
+                    System.out.println("☹ OOPS!!! I'm sorry, but I don't know what that means :-( Please enter something else!");
+                    System.out.println("____________________________________________________________");
+                    continue;
+                }
+                String taskDescription;
+                try {
+                    taskDescription = extractDescription(userInput);
+                } catch (DukeException e) {
+                    System.out.println("____________________________________________________________");
+                    System.out.println("☹ OOPS!!! The description of a " + taskType + " cannot be empty.");
+                    System.out.println("____________________________________________________________");
+                    continue;
+                }
+                switch (taskType) {
+                case "todo":
+                    addTask(new Todo(taskDescription));
+                    break;
+                case "deadline":
```

Comment:
Consider refactoring magic strings such as "todo" and "deadline" into constants.